### DISCUSSION TITLE
Align CALL and function call syntax

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#404] [Ideas]
Currently, the REXX CALL syntax differs from function calls:
```
Call myproc 1,2,3 
say myproc(1,2,3)
```
To ensure consistency and improve usability, the syntax for ordinary calls should align with function calls. 
```
Call myproc 1,2,3 
Call myproc(1,2,3)   /* extended syntax */ 
say myproc(1,2,3)
```
This, by the way, was one of the first changes we introduced in BREXX: aligning the CALL syntax with function call syntax for consistency.

**Comment by mikebeer** [comment]
Good idea.
Thus call would be similar to a void in C.



From: Peter-Jacob ***@***.***>
Sent: Dienstag, 17. Dezember 2024 08:50
To: adesutherland/CREXX ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [adesutherland/CREXX] Align CALL and function call syntax (Discussion #404)


Currently, the REXX CALL syntax differs from function calls:

Call myproc 1,2,3

say myproc(1,2,3)

To ensure consistency and improve usability, the syntax for ordinary calls should align with function calls.

Call myproc 1,2,3

Call myproc(1,2,3)   /* extended syntax */

say myproc(1,2,3)

This, by the way, was one of the first changes we introduced in BREXX: aligning the CALL syntax with function call syntax for consistency.
